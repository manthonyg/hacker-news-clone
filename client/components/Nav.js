import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const NavList = styled.ul`
  list-style: none;
  text-align: center;
  float: left;
`;

const NavItem = styled.li`
  display: inline-block;
  padding: 1em;
`;

const NavContainer = styled.div`
  postion: fixed;
  height: 8vh;
  width: 100%;
  background-color: #eee;
  margin: 0;
  padding: 0;
`;

const NavButton = styled.button`
  border: 2px solid #ccc;
  width: 50px;
  height: 50px;
  float: right;
`;
function Nav() {
  return (
    <NavContainer>
      <NavButton />
      <NavList>
        <NavItem>
          <Link to="/">Top</Link>
        </NavItem>
        <NavItem>
          <Link to="/new">New</Link>
        </NavItem>
      </NavList>
    </NavContainer>
  );
}

export default Nav;
