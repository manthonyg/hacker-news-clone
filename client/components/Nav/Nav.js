import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const NavList = styled.ul`
  list-style: none;
  text-align: center;
  float: left;
`;

const NavItem = styled.li`
  display: inline-block;
  padding: 1em;
`;

const NavContainer = styled.div`
  postion: fixed;
  height: 10vh;
  width: 100%;
  background-color: ${props => {
    if (props.theme.theme === 'light') return '#f1f1f1';
    return '#212121';
  }};
  margin: 0;
  padding: 0;
`;

const NavButton = styled.button`
  border: 2px solid #ccc;
  background-color: ${props => {
    if (props.theme.theme === 'light') return '#f1f1f1';
    return '#121212';
  }};
  color: ${props => {
    if (props.theme.theme === 'light') return '#121212';
    return '#f1f1f1';
  }};
  width: 50px;
  height: 50px;
  float: right;
`;
function Nav({ onClick, theme }) {
  return (
    <NavContainer>
      <NavButton onClick={onClick}>
        {theme.theme === 'light' ? 'dark mode' : 'light mode'}
      </NavButton>
      <NavList>
        <NavItem>
          <Link to="/">Top</Link>
        </NavItem>
        <NavItem>
          <Link to="/new">New</Link>
        </NavItem>
      </NavList>
    </NavContainer>
  );
}

export default Nav;
