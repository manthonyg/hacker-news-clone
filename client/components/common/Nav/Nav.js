import React from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";
import Heading from "../Heading/Heading";

const NavItem = styled.li`
  display: inline-block;
  padding: 0px 10px;
`;

const NavContainer = styled.div`
  postion: fixed;
  display: flex;
  align-items: center;
  height: 100%;
  width: 100%;
  background-color: ${(props) => {
    if (props.theme.theme === "light") return "#f1f1f1";
    return "#181818";
  }};
  margin: 0;
  padding: 0;
`;

const StyledLink = styled(Link)`
  text-decoration: none;
  font-weight: 800;
`;

const NavButton = styled.button`
  font-size: 1.3rem;
  padding: 0px;
  border: none;
  background: none;
  float: right;
`;

function Nav({ onClick, theme }) {
  return (
    <NavContainer>
      <NavButton onClick={onClick}>
        {theme.theme === "light" ? "ðŸŒƒ" : "ðŸŒ†"}
      </NavButton>
      <NavItem>
        <Heading underlined p>
          HNC
        </Heading>
      </NavItem>
      <NavItem>
        <StyledLink to="/">
          <Heading h5>top</Heading>
        </StyledLink>
      </NavItem>
      <NavItem>
        <StyledLink to="/best">
          <Heading h5>best</Heading>
        </StyledLink>
      </NavItem>
      <NavItem>
        <StyledLink to="/new">
          <Heading h5>new</Heading>
        </StyledLink>
      </NavItem>
    </NavContainer>
  );
}

export default Nav;
