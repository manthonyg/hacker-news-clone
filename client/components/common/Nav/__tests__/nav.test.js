import React from "react";
import renderer from "react-test-renderer";
import Nav from "../Nav";
import RenderWithRouter from "../../../../test_utils/renderWithRouter";

it("renders correctly", () => {
  const testTheme = { color: "red" };
  const tree = renderer
    .create(
      <RenderWithRouter>
        <Nav theme={testTheme}></Nav>
      </RenderWithRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
