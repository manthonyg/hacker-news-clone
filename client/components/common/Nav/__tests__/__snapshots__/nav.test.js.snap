// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxheI eLlhWc"
>
  <button
    className="sc-AxmLO ctbvAW"
    onClick={[Function]}
  >
    ðŸ’¡
  </button>
  <li
    className="sc-AxgMl WXvjg"
  >
    <h1
      className="sc-AxjAm gijxXQ"
    >
      HNC
    </h1>
  </li>
  <li
    className="sc-AxgMl WXvjg"
    data-testid="NAV_LINK_TOP"
  >
    <a
      className="sc-Axmtr jzWgds"
      href="/"
      onClick={[Function]}
    >
      <h5
        className="sc-AxhUy iTChLH"
      >
        top
      </h5>
    </a>
  </li>
  <li
    className="sc-AxgMl WXvjg"
    data-testid="NAV_LINK_BEST"
  >
    <a
      className="sc-Axmtr jzWgds"
      href="/best"
      onClick={[Function]}
    >
      <h5
        className="sc-AxhUy iTChLH"
      >
        best
      </h5>
    </a>
  </li>
  <li
    className="sc-AxgMl WXvjg"
    data-testid="NAV_LINK_NEW"
  >
    <a
      className="sc-Axmtr jzWgds"
      href="/new"
      onClick={[Function]}
    >
      <h5
        className="sc-AxhUy iTChLH"
      >
        new
      </h5>
    </a>
  </li>
</div>
`;
